cmd/world/cardinal/start.go

```diff
@@ -100,42 +103,41 @@ This will start the following Docker services and its dependencies:
 		}
 
 		runEditor, err := cmd.Flags().GetBool(flagEditor)
-		if err == nil && runEditor {
-			// Find an unused port for the Cardinal Editor
-			cardinalEditorPort, findPortError := findUnusedPort(cePortStart, cePortEnd)
-			if findPortError == nil {
-				cmdBlue := lipgloss.NewStyle().Foreground(lipgloss.Color("6"))
-				fmt.Println(cmdBlue.Render("Preparing Cardinal Editor..."))
-				fmt.Println(cmdBlue.Render(fmt.Sprint("Cardinal Editor will be run on localhost:", cardinalEditorPort)))
-				cePrepChan := make(chan struct{})
-				go func() {
-					err := runCardinalEditor(cfg.RootDir, cfg.GameDir, cardinalEditorPort, cePrepChan)
-					if err != nil {
-						cmdStyle := lipgloss.NewStyle().Foreground(lipgloss.Color("11"))
-						fmt.Println(cmdStyle.Render("Warning: Failed to run Cardinal Editor"))
-						logger.Error(eris.Wrap(err, "Failed to run Cardinal Editor"))
-
-						// continue if error
-						cePrepChan <- struct{}{}
-					}
-				}()
-				// Waiting cardinal editor preparation
-				<-cePrepChan
-			} else {
-				// just log the error if the editor fails to run
-				logger.Error(eris.Wrap(findPortError, "Failed to find an unused port for Cardinal Editor"))
-			}
-		} else {
-			// just log the error if the editor fails to run
-			logger.Error(eris.Wrap(err, "Failed to run Cardinal Editor"))
+		if err != nil {
+			return err
 		}
 
 		fmt.Println("Starting Cardinal game shard...")
 		fmt.Println("This may take a few minutes to rebuild the Docker images.")
 		fmt.Println("Use `world cardinal dev` to run Cardinal faster/easier in development mode.")
 
-		err = teacmd.DockerStartAll(cfg)
-		if err != nil {
+		group, ctx := errgroup.WithContext(cmd.Context())
+
+		// Start the World Engine stack
+		group.Go(func() error {
+			if err := teacmd.DockerStartAll(cfg); err != nil {
+				return eris.Wrap(err, "Encountered an error with Docker")
+			}
+			return eris.Wrap(ErrGracefulExit, "Stack terminated")
+		})
+
+		// Start Cardinal Editor is flag is set to true
+		if runEditor {
```

Comment:
another small nit.
on `world cardinal start` there is no information about which port that cardinal editor used, it will be better if the information is printed like on `world cardinal dev`.

Reply:
Let's do that in a separate PR

Reply:
ok